src/lsst/cmservice/routers/v2/campaigns.py

```diff
@@ -282,8 +283,8 @@ async def update_campaign_resource(
     # If the patch data is requesting a status change, we will not affect that
     # directly, but defer it to a background task
     if patch_data.status is not None:
-        # TODO implement middleware to assign a request_id to every request
-        request_id = uuid4()
+        if (request_id := correlation_id.get()) is None:
+            raise HTTPException(status_code=500, detail="Cannot patch resource without a X-Request-Id")
```

Comment:
I don't follow why the patch operation in particular needs the request ID, any more than the other endpoints. Is this an error path that's mostly-impossible because fastapi should always be supplying the request ID?

Reply:
It's because for a campaign, the patch route is what implements the start-stop state-change commands, and this request id is wired to the activity log record. I think this is the only case where the request ID is materialized and mandatory in a route, but it's omnipresent for correlation logging.